ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/toolbarview/MessengerToolbarFragmentTest.java

```diff
@@ -0,0 +1,570 @@
+        when(view.findViewById(R.id.ko__messenger_toolbar_back_button)).thenReturn(view);
+        when(view.findViewById(R.id.ko__messenger_toolbar_title)).thenReturn(view);
+        when(view.findViewById(R.id.ko__messenger_toolbar_subtitle)).thenReturn(view);
+        when(view.findViewById(R.id.ko__messenger_toolbar_avatar1)).thenReturn(view);
+        when(view.findViewById(R.id.ko__messenger_toolbar_avatar2)).thenReturn(view);
+        when(view.findViewById(R.id.ko__messenger_toolbar_avatar3)).thenReturn(view);
+        when(view.getTransitionName()).thenReturn(TEST_TEXT);
+        when(fragmentActivity.isFinishing()).thenReturn(false);
+        when(handler.getLooper()).thenReturn(looper);
+
+        PowerMockito.whenNew(MessengerToolbarCollapsedFragment.class).withNoArguments().thenReturn(messengerToolbarCollapsedFragment);
+        PowerMockito.whenNew(MessengerToolbarExpandedFragment.class).withNoArguments().thenReturn(messengerToolbarExpandedFragment);
+
+        PowerMockito.mockStatic(Fragment.class);
+        PowerMockito.mockStatic(MessengerToolbarFactory.class);
+        PowerMockito.mockStatic(MessengerTemplateHelper.class);
+        PowerMockito.mockStatic(Looper.class);
+        PowerMockito.mockStatic(Build.VERSION.class);
+
+        Whitebox.setInternalState(android.os.Build.VERSION.class, SDK_INT, Build.VERSION_CODES.LOLLIPOP);
+
+        PowerMockito.when(Looper.class, MY_LOOPER_METHOD).thenReturn(looper);
+        PowerMockito.when(MessengerToolbarFactory.class, GET_PRESENTER_METHOD, messengerToolbarFragment).
+                thenReturn(presenter);
+        PowerMockito.when(fragmentHostCallback, GET_ACTIVITY_METHOD).thenReturn(fragmentActivity);
+        PowerMockito.when(fragmentHostCallback, GET_HANDLER_METHOD).thenReturn(handler);
+        PowerMockito.when(messengerToolbarCollapsedFragment, IS_ADDED_METHOD).thenReturn(true);
+        PowerMockito.when(messengerToolbarCollapsedFragment, GET_VIEW_METHOD).thenReturn(view);
+        PowerMockito.when(messengerToolbarExpandedFragment, IS_ADDED_METHOD).thenReturn(true);
+        PowerMockito.when(messengerToolbarExpandedFragment, GET_VIEW_METHOD).thenReturn(view);
+
+        PowerMockito.suppress(PowerMockito.methods(Fragment.class, ON_CREATE_METHOD));
+        PowerMockito.suppress(PowerMockito.methods(Fragment.class, ON_VIEW_CREATED_METHOD));
+        PowerMockito.suppress(PowerMockito.methods(Fragment.class, ON_ACTIVITY_CREATED_METHOD));
+        PowerMockito.suppress(PowerMockito.methods(Fragment.class, ON_DESTROY_VIEW_METHOD));
+        PowerMockito.suppress(PowerMockito.methods(MessengerTemplateHelper.class, APPLY_BACKGROUND_THEME_METHOD));
+    }
+
+    @Test
+    public void onCreate() {
+        // Arrange & Act
+        messengerToolbarFragment.onCreate(bundle);
+    }
+
+    @Test
+    public void onCreateView() {
+        // Arrange & Act
+        View viewResult = messengerToolbarFragment.onCreateView(inflater, container, bundle);
+
+        // Assert
```

Comment:
✅ Good to see that the AAA("Arrange-Act-Assert") arranging and formatting pattern is followed.

Some of the main advantages are that it clearly separates what is being tested from the setup and verification steps. It also clarifies and focuses attention on a historically successful and generally necessary set of test steps.

**No change suggested, looks good!**	  

<a id="8821d480-4f95-11e8-bb12-7de6652163a5" href="none" codereview-uuid-reserved-tag />